src/shared/components/download/__tests__/Download.spec.tsx

```diff
@@ -90,18 +95,37 @@ describe('Download component', () => {
     [FileFormat.tsv, true],
   ])(
     'should show column selection component when %s file type is selected and otherwise hide it',
-    (value, columnSelect) => {
+    async (value, columnSelect) => {
       const formatSelect = screen.getByTestId('file-format-select');
       fireEvent.change(formatSelect, { target: { value } });
-      const customise = screen.queryByText('Customize data');
-      if (columnSelect) {
-        expect(customise).toBeInTheDocument();
-      } else {
-        expect(customise).not.toBeInTheDocument();
-      }
+      await waitFor(() => {
```

Comment:
Why is this needed? What's the difference with before that now requires this to be wrapped by waitFor?

Reply:
The change event triggers a state change, so the elements might not be there at the time of check. I'm not sure it worked as expected before?

Reply:
I'm wondering trying to reuse `customise` variable makes it a bit more complex that it needs to be. Seeing that bit of test looks to me like this bit of the article https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-waitfor-to-wait-for-elements-that-can-be-queried-with-find

Reply:
So I could switch to `await screen.findByText('Customize data');` and then check for `null` instead of `toBeInTheDocument`.

Reply:
Actually it's not great is it? It shouldn't be waiting for something we know shouldn't be there... 🤔

Reply:
True. Is there a better question to ask the DOM? Under what condition we are ready to expect that something isn't there not just because it didn't had time to appear, but because it will not be there even after the next render or whatever

Reply:
So actually it turns out the async/await wasn't needed so I've reverted it back to how it was.